File: app/lib/screens/project_details/models/task.dart
class Tasks {
  int? id;
  int? projectId;
  String? title;
  String? description;
  int? status;
  DateTime? dueDate;
  int? createdBy;
  DateTime? createdDate;

  Tasks({
    this.id,
    this.projectId,
    this.title,
    this.description,
    this.status,
    this.dueDate,
    this.createdBy,
    this.createdDate,
  });

  factory Tasks.fromJson(Map<String, dynamic> json) {
    return Tasks(
      id: json['id'],
      projectId: json['project_id'],
      title: json['title'],
      description: json['description'],
      status: json['status'],
      dueDate: DateTime.parse(json['due_date']),
      createdBy: json['created_by'],
      createdDate:  DateTime.parse(json['created_date']),
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'project_id': projectId,
    'title': title,
    'description': description,
    'status': status,
    'due_date': dueDate?.toIso8601String(),
    'created_by': createdBy,
    'created_date': createdDate,
  };
  /// ✅ This is your missing method:
  Tasks copyWith({
    int? id,
    int? projectId,
    String? title,
    String? description,
    int? status,
    DateTime? dueDate,
    int? createdBy,
    DateTime? createdDate,
  }) {
    return Tasks(
      id: id ?? this.id,
      projectId: projectId ?? this.projectId,
      title: title ?? this.title,
      description: description ?? this.description,
      status: status ?? this.status,
      dueDate: dueDate ?? this.dueDate,
      createdBy: createdBy ?? this.createdBy,
      createdDate: createdDate ?? this.createdDate,
    );
  }

}

File: api/SprintoApi/SprintoApi/Models/Task.cs
using System.ComponentModel.DataAnnotations;

namespace SprintoApi.Models
{
    public class Tasks
    {
        [Key]
        public int id { get; set; }
        public int project_id { get; set; }
        public string Description { get; set; }
        public string title { get; set; }
        public int status { get; set; }
        public DateTime due_date { get; set; }
        public int created_by { get; set; }
        public DateTime created_date { get; set; }

    }

    public class TasksDto
    {
        public int id { get; set; }
        public int project_id { get; set; }
        

    }

    public class UpdateTaskStatusRequest
    {
        public int Id { get; set; }
        public int Status { get; set; }
    }
}
